src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -705,6 +705,8 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(TokenLogMessages.IDX10209, token.Length, MaximumTokenSizeInBytes)));
 
             var tokenParts = token.Split(new char[] { '.' }, JwtConstants.MaxJwtSegmentCount + 1);
+            ClaimsPrincipal claimsPrincipal = null;
```

Comment:
Not need for redundant assignment. I'd move these variables closer to where they are used.